packages/react-native-performance/src/performance-observer.ts

```diff
@@ -52,14 +70,17 @@ export const createPerformanceObserver = ({
       }
     }
 
-    receiveRecord = entry => {
+    receiveRecord = (entry: PerformanceEntry) => {
       if (this.entryTypes.has(entry.entryType)) {
         this.buffer.push(entry);
         this.scheduleEmission();
       }
     };
 
-    observe(options) {
+    observe(options: ObserveOptionType1): void;
```

Comment:
I wonder if we can't rewrite this to `options: ObserveOptionType1 | ObserveOptionType2`.

Reply:
if we make it that way, it will combine them together. It will be like 
`{ entryTypes: EntryType[], types: EntryType, buffered: boolean } `
So, itt will not give hint to user correct. It will accept for example 
`{ entryTypes: ['resources'], buffered: false }`  but if we use entryTypes we can't use buffered. That is why i used overload here.

Reply:
Maybe my TS-fu is lacking, but isn't that if you do `&`?

Reply:
If you do & it should have all of it.

Reply:
I think you're wrong about the `|` operator: https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#union-types

Reply:
If we put | instead of overload, it will accept all props options in the place that we use in this case App.tsx in Vanilla example project.
You can try out here: https://www.typescriptlang.org/play?#code/C4TwDgpgBAogdsATiAKuaBeKBnJBLOAcwChjRIoB5AI2wkQDcJKxg8B7ONSARiiwDexKFAgJk3CNgBcscanQBtALoBuYgF91ZdFVr0mLNp0kAmflCEixSBZFnxbk9SOoBXAGYf6EACYB+WWp2dgAbCABDOHUNUggADzB2RGAoAGNQiOxsKAAFeg9kgFsotIgAGXZCQnoAdTxgAAt2N2BKJkRQ9gjfS2FReIiisHCACgBKPpERJrxsADp2fUYIUYEbCXRZAHIikABlfCJtgBoody8fX1kPCNC6M427KVkVDXGXKFj+pboV0fYrA4cBkej+hiBJnQfAAPmCDMxIVx0KZJkJ+t9vgkkil0plsnkCsVShUqjVEPUmu16F0elMBkMRqs0f0Zo05otlkw1k9JDs9odEARCKdzp5vIg-Dc7g9RPJJKC3h8MaQRL8EQCkaCaODEcZkbxxrIGOw8L5Pur-oD9dquXrgWYjVATWbtGq7ZqbbIoiAWSrNKQgA